CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+licensing restrictions.
+
+### Code of Conduct
+
+Before contributing please read the [code of conduct](./CODE_OF_CONDUCT.md)
+
+Note that our [code of conduct](./CODE_OF_CONDUCT.md) applies to all platforms
+and venues related to this project; please follow it in all your interactions
+with the project and its participants.
+
+### Contributor License Agreement
+
+Keep in mind that when you submit your Pull Request, you'll need to sign the
+CLA via the click-through using CLA-Assistant. If you'd like to execute our
+corporate CLA, or if you have any questions, please drop us an email at
+opensource@newrelic.com.
+
+For more information about CLAs, please check out Alex Russell’s excellent
+post, [“Why Do I Need to Sign
+This?”](https://infrequently.org/2008/06/why-do-i-need-to-sign-this/).
+
+### Slack
+
+We host a public Slack with a dedicated channel for contributors and
+maintainers of open source projects hosted by New Relic. If you are
+contributing to this project, you're welcome to request access to the
+\#oss-contributors channel in the newrelicusers.slack.com workspace. To request
+access, see https://newrelicusers-signup.herokuapp.com/.
+
+## PR Guidelines
+
+### Version Support
+
+When contributing, keep in mind that New Relic customers (that's you!) are running many different versions of Node, some of them pretty old. Changes that depend on the newest version of Node will probably be rejected, especially if they replace something backwards compatible.
+
+Be aware that the instrumentation needs to work with a wide range of versions of the instrumented modules, and that code that looks nonsensical or overcomplicated may be that way for compatibility-related reasons. Read all the comments and check the related tests before deciding whether existing code is incorrect.
+
+If you’re planning on contributing a new feature or an otherwise complex contribution, we kindly ask you to start a conversation with the maintainer team by opening up an Github issue first. 
+
+### General Guidelines
+
+In general, we try to limit adding third-party production dependencies. If one is necessary, please be prepared to make a clear case for the need.
+
+### Coding Style Guidelines/Conventions
+
+We use eslint to enforce certain coding standards. Please see our [.eslintrc](./.eslintrc.js) file for specific rule configuration.
+
+### Testing Guidelines
+
+The MySql instrumentation module includes a suite of functional tests which should be used to
+verify your changes don't break existing functionality.
+
+Functional tests against specific versions of instrumented modules are stored
+in `test/versioned/`. They are written in 
+[node-tap](https://github.com/isaacs/node-tap), and have the extension `.tap.js`.
+
+#### Running Tests
+
+To run the test suite locally, you will need a Docker container running MySQL as setup with the [New Relic Node Agent](https://github.com/newrelic/node-newrelic/blob/main/package.json#L131).
```

Comment:
I'd prob get more specific about how to get it setup. 

It looks like we have pretty good instructions on the Node agent side, so perhaps saying "you can follow the setup instructions here: https://github.com/newrelic/node-newrelic/blob/main/CONTRIBUTING.md#setup".